app/controllers/gi_controller.rb

```diff
@@ -11,6 +11,8 @@ def client
   end
 
   def scrubbed_params
-    params.except(:action, :controller, :format)
+    params.except(:action, :controller, :format).transform_values do |v|
+      v.encode!('UTF-8', invalid: :replace, undef: :replace, replace: '')
```

Comment:
In other words, i don't know if we want to be "sanitizing" or replacing components in the params so much as validating them and raising a helpful error in the event that there is an issue.